CMakeLists.txt

```diff
@@ -159,75 +166,116 @@ set(XWIDGETS_SOURCES
+                          PUBLIC xproperty)
 
-set_target_properties(xwidgets PROPERTIES
-    CXX_EXTENSIONS OFF
-    CXX_STANDARD_REQUIRED 14)
+    set_target_properties(${target_name} PROPERTIES
+                          PUBLIC_HEADER "${XWIDGETS_HEADERS}"
+                          COMPILE_DEFINITIONS "XWIDGETS_EXPORTS"
+                          PREFIX ""
+                          VERSION ${XWIDGETS_BINARY_VERSION}
+                          SOVERSION ${XWIDGETS_BINARY_CURRENT}
+                          OUTPUT_NAME "lib${output_name}")
 
-target_compile_features(xwidgets PRIVATE cxx_std_14)
+    # Compilation flags
+    # =================
 
-if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Intel")
-    target_compile_options(xwidgets PUBLIC -Wunused-parameter -Wextra -Wreorder)
-    if (NOT DISABLE_ARCH_NATIVE)
-        target_compile_options(xwidgets PUBLIC -march=native)
-    endif()
+    string(TOUPPER "${CMAKE_BUILD_TYPE}" U_CMAKE_BUILD_TYPE)
+
+    set_target_properties(${target_name} PROPERTIES
+        CXX_EXTENSIONS OFF
+        CXX_STANDARD_REQUIRED 14)
 
-    # Enable link time optimization and set the default symbol
-    # visibility to hidden (very important to obtain small binaries)
-    if (NOT ${U_CMAKE_BUILD_TYPE} MATCHES DEBUG)
-        # Check for Link Time Optimization support
-        # (GCC/Clang)
-        CHECK_CXX_COMPILER_FLAG("-flto" HAS_LTO_FLAG)
-        if (HAS_LTO_FLAG)
-            target_compile_options(xwidgets PUBLIC -flto)
+    target_compile_features(${target_name} PRIVATE cxx_std_14)
+
+    if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+        target_compile_options(${target_name} PUBLIC -Wunused-parameter -Wextra -Wreorder)
+        if (NOT DISABLE_ARCH_NATIVE)
+            target_compile_options(${target_name} PUBLIC -march=native)
         endif()
 
-        # Intel equivalent to LTO is called IPO
-        if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
-            CHECK_CXX_COMPILER_FLAG("-ipo" HAS_IPO_FLAG)
-            if (HAS_IPO_FLAG)
-                target_compile_options(xwidgets PUBLIC -ipo)
+        # Enable link time optimization and set the default symbol
+        # visibility to hidden (very important to obtain small binaries)
+        if (NOT ${U_CMAKE_BUILD_TYPE} MATCHES DEBUG)
+            # Check for Link Time Optimization support
+            # (GCC/Clang)
+            CHECK_CXX_COMPILER_FLAG("-flto" HAS_LTO_FLAG)
+            if (HAS_LTO_FLAG)
+                target_compile_options(${target_name} PUBLIC -flto)
+            endif()
+
+            # Intel equivalent to LTO is called IPO
+            if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+                CHECK_CXX_COMPILER_FLAG("-ipo" HAS_IPO_FLAG)
+                if (HAS_IPO_FLAG)
+                    target_compile_options(${target_name} PUBLIC -ipo)
+                endif()
```

Comment:
I think we should completely remote the section about LTO/IPO from the cmake.

Reply:
Ok! (I had kept it as I did not know what was for)